File: lotteryticket/src/main/java/com/wuhai/lotteryticket/ui/holder/base/BaseDataBindingHolder.java
package com.wuhai.lotteryticket.ui.holder.base;

import android.content.Context;
import android.view.View;

import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

/**
 * 作者 wuhai
 *
 * 创建日期 2018/5/25 9:54
 *
 * 描述：
 */

public abstract class BaseDataBindingHolder<T extends Object> extends RecyclerView.ViewHolder  {


    protected   T mData;
    protected Context mContext;

    public BaseDataBindingHolder(View itemView) {
        super(itemView);
    }

    public BaseDataBindingHolder(ViewDataBinding binding) {
        super(binding.getRoot());
    }
    public BaseDataBindingHolder(ViewDataBinding binding, Context context) {
        super(binding.getRoot());
        mContext = context;
    }

    public BaseDataBindingHolder(View itemView, Context context) {
        super(itemView);
        mContext = context;
    }

    public void setmData(T mData) {
        this.mData = mData;
    }

    public abstract void refreshView();
}

File: lotteryticketkotlin/src/main/java/com/wuhai/demo/lotteryticketkotlin/ui/holder/base/BaseDataBindingHolder.kt
package com.wuhai.demo.lotteryticketkotlin.ui.holder.base

import android.content.Context
import android.view.View
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

/**
 * 作者 wuhai
 *
 * 创建日期 2018/5/25 9:54
 *
 * 描述：
 */
abstract class BaseDataBindingHolder<T : Any?> : RecyclerView.ViewHolder {
    protected var mData: T? = null
    protected var mContext: Context? = null

    constructor(itemView: View?) : super(itemView!!) {}
    constructor(binding: ViewDataBinding) : super(binding.root) {}
    constructor(binding: ViewDataBinding, context: Context?) : super(binding.root) {
        mContext = context
    }

    constructor(itemView: View?, context: Context?) : super(itemView!!) {
        mContext = context
    }

    fun setmData(mData: T) {
        this.mData = mData
    }

    abstract fun refreshView()
}